### PR TITLE
fix: Issue with QR Code not being seen with MS Authenticator

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes
MS Authenticator does not recognize QR codes from the Authenticator

#### Issue #, if available

#1837 

#### Description of how you validated changes
Manually tested and Jest test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by calebpollman — COMMENTED** [review]
LGTM but can we add a unit test?

**Review by zchenwei — APPROVED** [review]
LGTM

File: packages/react/src/components/Authenticator/SetupTOTP/__tests__/SetupTOTP.test.tsx [hunk 1]
```diff
@@ -88,4 +88,28 @@ describe('SetupTOTP', () => {
     expect(toDataURLSpy).toHaveBeenCalledTimes(1);
     expect(toDataURLSpy).toHaveBeenCalledWith(customTotpCode);
   });
+
+  it('handles customTotpIssuer with spaces', async () => {
```
Comment (calebpollman) [inline packages/react/src/components/Authenticator/SetupTOTP/__tests__/SetupTOTP.test.tsx:92]: Nit: could just place this test its own `describe` block and remove the component code, might make it easier for future readers:
```
describe('getTotpCode', () => {
  it('handles customTotpIssuer with spaces', () => {
    const customTotpIssuer = 'customTOTPIssuer spaces';
    const customTotpUsername = 'customTotpUsername';

    const customTotpCode = getTotpCode(
      customTotpIssuer,
      customTotpUsername,
      SECRET_KEY
    );

    expect(customTotpCode).toBe(
      'otpauth://totp/customTOTPIssuer%20spaces:customTotpUsername?secret=secretKey&issuer=customTOTPIssuer%20spaces'
    );
  });
});
```

**Review by calebpollman — APPROVED** [review]
Thanks for adding the unit test!

File: packages/react/src/components/Authenticator/SetupTOTP/__tests__/SetupTOTP.test.tsx (hunk 1 above)
Comment (ErikCH) [reply packages/react/src/components/Authenticator/SetupTOTP/__tests__/SetupTOTP.test.tsx:92]: OK I'll move it out into a new describe

**Pull request merged**